crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -345,20 +345,24 @@ pub fn export_schema() -> String {
     schema_builder().finish().sdl()
 }
 
+pub(crate) struct QueryString(pub String);
+
 async fn graphql_handler(
     ConnectInfo(addr): ConnectInfo<SocketAddr>,
     schema: axum::Extension<SuiGraphQLSchema>,
     headers: HeaderMap,
     req: GraphQLRequest,
 ) -> (axum::http::Extensions, GraphQLResponse) {
     let mut req = req.into_inner();
+    let query = QueryString(req.query.clone());
```

Comment:
would this capture variables? we have similar logic in `logger.rs`, where we log `ctx.stringify_execute_doc(&document, variables)` if specified in config. But for the timeout extension i think it'd need to hook into `request` or `execute` so we can correctly time out while waiting for the response. I'm not sure if there's a way to access that same data in the `execute` hook, or if it would be a code smell to have `logger` somehow populate global context with that information for the `timeout` extension to use...

Reply:
Good point, let me think more and see what I can do.

Reply:
```
sui_graphql_rpc::extensions::timeout: query_id=44d9f1ec-e4df-488c-bd2a-a0b31d5bb203 session_id=127.0.0.1:59043 error_code="REQUEST_TIMEOUT" query=query { a:transactionBlocks(first: 10) { nodes { digest } } b:transactionBlocks(first: 50) { nodes { digest } } }
```